### PR TITLE
feature/william_diaz/solid

### PR DESCRIPTION
**Pull request by WilliamrDiaz**
William Rodrigo Diaz Maca

### CODE REVIEW

File: principles/solid/README.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
 ### SOLID Principles
+It is an acronym and each letter is a different principle. This principle is very popular when an architecture needs to be implemented in any type of project.
```
Comment (zahydo) [inline principles/solid/README.md]: Remove this change

File: principles/solid/dependencyInjection.md [hunk 2]
```diff
@@ -1,14 +1,44 @@
...
-[TODO]
+It depends on an abstraction and not on something concrete. It defines that an element must depend on an abstraction and not on something concrete. it is necessary that the code is much more reusable, much more generic so that the code can have as many forms as we want. 
 
 ### Bad example
 
-[TODO]
+In an application, a view presenter class has a direct dependency on the data model class. This makes the presenter very dependent on the data model and difficult to test.
+
+```Kotlin
+class ViewPresenter {
```
Comment (zahydo) [inline principles/solid/dependencyInjection.md]: You should use Java

File: principles/solid/interfacesegregation.md [hunk 3]
```diff
@@ -1,14 +1,57 @@
...
 
 ### Good example:
 
-[TODO]
+Interfaces can be divided into two: one for saving and loading data and one for sending e-mails. In this way, clients only implement the interfaces they need.
+
+```Kotlin
+interface DataSaver {
+    fun saveData(Object data);
+    fun loadData();
```
Comment (zahydo) [inline principles/solid/interfacesegregation.md]: Use Java

File: principles/solid/liskvsustitution.md [hunk 4]
```diff
@@ -1,14 +1,60 @@
 ## Liskov Substitution
 
-[TODO]
+We should be able to use a child class to replace a parent class without getting errors.
```
Comment (zahydo) [inline principles/solid/liskvsustitution.md]: The description of the principles is very simple. The principle should have the problem and how the principle solves it.

**Review by zahydo — CHANGES_REQUESTED** [review]
It's missing the Single Responsibility principle

File: principles/solid/README.md (hunk 1 above)
Comment (WilliamrDiaz) [reply principles/solid/README.md]: Done!

File: principles/solid/dependencyInjection.md (hunk 2 above)
Comment (WilliamrDiaz) [reply principles/solid/dependencyInjection.md]: Fix, thanks!

File: principles/solid/interfacesegregation.md (hunk 3 above)
Comment (WilliamrDiaz) [reply principles/solid/interfacesegregation.md]: Fix, thanks!

File: principles/solid/liskvsustitution.md (hunk 4 above)
Comment (WilliamrDiaz) [reply principles/solid/liskvsustitution.md]: Done!

File: principles/solid/dependencyInjection.md [hunk 5]
```diff
@@ -1,14 +1,44 @@
...
 
-[TODO]
+You can create an interface to the data model and make the presenter depend on the interface instead of the particular data model class. In this way, you can provide a different implementation of the data model for different use cases and you can easily test the presenter with a test data model.
+
+```Java
+interface DataModel {
+    // Métodos del modelo de datos
+}
+
+class ViewPresenter(val dataModel: DataModel) {
```
Comment (zahydo) [inline principles/solid/dependencyInjection.md]: This is not Java, this is Kotlin, don't copy codes!

File: principles/solid/interfacesegregation.md [hunk 6]
```diff
@@ -1,14 +1,57 @@
...
+In an application, an interface that defines methods for saving and loading data also includes a method for sending e-mails. If a client only needs the methods for saving and loading data, it is forced to implement the method for sending e-mails.
+
+```Java
+interface DataManager {
+    fun saveData(data: Any)
+    fun loadData()
+    fun sendEmail(recipient: String, message: String)
+}
+
+class MyClient : DataManager {
```
Comment (zahydo) [inline principles/solid/interfacesegregation.md]: The same with the rest of the examples. Don't copy code

File: principles/solid/dependencyInjection.md (hunk 5 above)
Comment (WilliamrDiaz) [reply principles/solid/dependencyInjection.md]: I'm sorry! I just changed Kotlin for Java and I didn't pay attention to the code syntax haha.
Fixed!

File: principles/solid/interfacesegregation.md (hunk 6 above)
Comment (WilliamrDiaz) [reply principles/solid/interfacesegregation.md]: Fixed!

**Review by zahydo — CHANGES_REQUESTED** [review]
The related principles are missing for all the principles.

**Comment by zahydo** [comment]
Checked

**Pull request closed without merging**